scripts/abins/atomsdata.py

```diff
@@ -92,15 +92,23 @@ def _check_item(item: _AtomData, n_atoms: Optional[int] = None) -> _AtomData:
             raise ValueError("Invalid structure of the dictionary to be added.")
 
         # "symbol"
-        if (symbol := item["symbol"]) not in abins.constants.ALL_SYMBOLS:
-            # Check is symbol was loaded as type bytes
-            if isinstance(symbol, bytes):
-                utf8_symbol = symbol.decode("utf-8")
+        if (symbol_raw := item["symbol"]) not in abins.constants.ALL_SYMBOLS:
+            # Check if symbol was loaded as type bytes
+            if isinstance(symbol_raw, bytes):
+                symbol = symbol_raw.decode("utf-8")
+            else:
+                symbol = symbol_raw
+                assert isinstance(symbol, str)
```

Comment:
perhaps a bit more informative error?
```python
elif isinstance(symbol, str):
    symbol = symbol_raw
else:
    raise TypeError(f"Invalid type for symbol {symbol_raw}")
```

Reply:
This `assert` is for mypy's benefit; it seems to be the only way to convince it that a variable that could once hold more than one type now holds a specific type.

I'm not sure that will work if it isn't a fairly simple `assert` statement.